jax/experimental/jax2tf/tests/tf_test_util.py

```diff
@@ -125,18 +129,24 @@ def run_tf(mode):
       else:
         assert False
 
-    def is_tf_exception(lim: jax2tf.jax2tf.Limitation):
+    def is_tf_exception(lim: jcs.Limitation):
       return (lim.ErrorType == 'Missing TF support' and
               self.tf_default_device.device_type in lim.Devices)
 
     result_tf = None
     for mode in ("eager", "graph", "compiled"):
-      current_limitations = jax2tf.jax2tf.all_limitations[:]
+      current_limitations = jcs.all_limitations[:]
+      # Monkey-patch jax2tf.TensorFlowTrace.get_primitive_impl to wrap the
+      # resulting primitive in a categorizer. We do that at every loop
+      # iteration because we want to restore the original implementation
+      # before returning, and the code inside the loop may throw exceptions
+      # and thus return early.
+      patch_get_primitive_impl()
```

Comment:
nit: I would just inline here the function here. I am not really sure though, for the cleanup you will need to define a function.

Reply:
Addressed in `6fc28be`. I actually put everything out of the loop now, since we can use `addCleanup`.